first-react-app/src/components/formComponents/SubmitForm.tsx

```diff
@@ -0,0 +1,167 @@
+      checkbox: boolean;
+      radio: null;
+      upload: null;
+    }>
+  ) {
+    const isTextWrong = errors.songName;
+    const isDateWrong = errors.date;
+    const isRadioWrong = errors.radio;
+
+    if (isTextWrong || isDateWrong || isRadioWrong) {
+      setpopUpClass("error-visible");
+      setpopUpText("You have to correct errors in form first only then you can submit.");
+    }
+  }
+
+  function handleData(data: {
+    songName: string;
+    date: string;
+    select: string;
+    checkbox: boolean;
+    radio: null;
+    upload: null | FileList;
+  }) {
+    let image = "";
+
+    try {
+      image = URL.createObjectURL(data.upload![0]);
+    } catch (error) {
+      image = "";
+    } finally {
+      const formData = [
+        data.songName,
+        data.date,
+        data.select,
+        data.checkbox.toString(),
+        data.radio as unknown as string,
+        image,
+      ];
+      setpopUpClass("confirm-visible");
+      setpopUpText("You have successfully created a card.");
+      props.isFormCorrect(formData);
+    }
+  }
+
+  return (
+    <form
+      onSubmit={handleSubmit((data) => {
+        handleData(data);
+      })}
+      className="submit-wrapper"
+    >
+      <p>Type in a name for the song</p>
+      <input
+        {...register("songName", {
+          required: true,
+          minLength: { value: 4, message: "Song name must be at least 4 symbols" },
+        })}
+      />
+      <p className="error">{errors.songName?.message}</p>
+      <p>When was this song created?</p>
+      <input
+        type="date"
+        {...register("date", {
+          required: true,
+          validate: (value) => {
+            const inputDate = new Date(value);
+            const currDate = new Date();
+            return inputDate > currDate ? "You cannot choose the future date" : true;
+          },
+        })}
+      />
+      <p className="error">{errors.date?.message}</p>
+      <p>Rate the song from 1 to 10</p>
+      <select
+        {...register("select", {
+          required: true,
+        })}
+      >
+        <option value="1">1</option>
+        <option value="2">2</option>
+        <option value="3">3</option>
+        <option value="4">4</option>
+        <option value="5">5</option>
+        <option value="6">6</option>
+        <option value="7">7</option>
+        <option value="8">8</option>
+        <option value="9">9</option>
+        <option value="10">10</option>
+      </select>
+      <label htmlFor="checkbox">
+        Does this song has implicit content?
+        <input id="checkbox" type="checkbox" {...register("checkbox")} />
+      </label>
+      <p>Was the song made by you or by someone else?</p>
+      <label htmlFor="radio-1">
+        By me
+        <input
+          id="radio-1"
+          value="By me"
+          type="radio"
+          {...register("radio", { required: "Need to choose one option" })}
+        />
+      </label>
+      <label htmlFor="radio-2">
```

Comment:
To explicitly associate a `<label>` element with an `<input>` element, you first need to add the id attribute to the `<input>` element. Next, you add the for attribute to the `<label>` element, where the value of for is the same as the id in the <input> element.

Alternatively, you can nest the `<input>` directly inside the `<label>`, in which case the for and id attributes are not needed because the association is implicit.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label
```suggestion
      <label>
```